.github/workflows/ci-cd.yml

```diff
@@ -33,11 +32,22 @@ jobs:
         run: |
           $defaultBranch = "${{ steps.data_gatherer.outputs.default_branch }}"
           $githubRef = "${{ github.ref }}"
+          $githubEventName = "${{ github.event_name }}"
           $isDefaultBranch = 'false'
+          $isPush = 'false'
+          $isPushToDefaultBranch = 'false'
           if ( $githubRef -like "*$defaultBranch*" ) {
             $isDefaultBranch = 'true'
           }
+          if ( $githubEventName -eq 'push' ) {
+            $isPush = 'true'
+          }
+          if ( $githubRef -like "*$defaultBranch*" -and $githubEventName -eq 'push' ) {
+            $isPushToDefaultBranch = 'true'
+          }
           Write-Output "::set-output name=is_default_branch::$(echo $isDefaultBranch)"
+          Write-Output "::set-output name=is_push::$(echo $isPush)"
+          Write-Output "::set-output name=is_push_to_default_branch::$(echo $isPushToDefaultBranch)"
```

Comment:
added two more conditional statements:
1. is_push = when the workflow is triggered by a push commit
2. is_push_to_default_branch = when the workflow is triggered by the a push commit in the default branch